### PR TITLE
Log file name, function name and line number

### LINKED ISSUE #343: Log entries for errors should contain function name for quicker debugging

### PR DESCRIPTION
**Pull request by prashanthpai**
Closes #343

### CODE REVIEW

**Comment by prashanthpai** [comment]
Sample output:

```log
DEBU[2017-12-01T15:33:37+05:30] registered gRPC method                        at="listener.go:peerrpc.(*Server).Serve:45" method="{RunStep false false}" service=transaction.TxnSvc
INFO[2017-12-01T15:33:37+05:30] Started GlusterD ReST server                  at="rest.go:rest.(*GDRest).Serve:94" ip:port="[::]:24007"
INFO[2017-12-01T15:33:37+05:30] started GlusterD SunRPC server                at="server.go:sunrpc.(*SunRPC).Serve:116" ip:port="[::]:24007"


ERRO[2017-12-01T15:34:26+05:30] volume not found                              at="store-utils.go:volume.GetVolume:53" volume=test

```

**Comment by kshlm** [comment]
> DEBU[2017-12-01T15:33:37+05:30] registered gRPC method                        at="listener.go:peerrpc.(*Server).Serve:45" method="{RunStep false false}" service=transaction.TxnSvc
> INFO[2017-12-01T15:33:37+05:30] Started GlusterD ReST server                  at="rest.go:rest.(*GDRest).Serve:94" ip:port="[::]:24007"
> INFO[2017-12-01T15:33:37+05:30] started GlusterD SunRPC server                at="server.go:sunrpc.(*SunRPC).Serve:116" ip:port="[::]:24007"
>
>
> ERRO[2017-12-01T15:34:26+05:30] volume not found                              at="store-utils.go:volume.GetVolume:53" volume=test

I don't like the source file and line being a part of the log context. The context should contain information that helps understand why the log was logged. I think it should be possible to write a custom log formatter that doesn't add the source info to the context, and instead adds it to the log metadata along with the time.

**Comment by prashanthpai** [comment]
New sample:

```log
[2017-12-01T18:00:00+05:30]  INFO rest.go:rest.(*GDRest).Serve:94: Started GlusterD ReST server ip:port=[::]:24007
[2017-12-01T18:00:00+05:30]  INFO server.go:sunrpc.(*SunRPC).Serve:116: started GlusterD SunRPC server ip:port=[::]:24007
```

File: pkg/logging/hook.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	n := runtime.Callers(6, pcs)
+	if n == 0 {
+		return nil
+	}
+
+	frames := runtime.CallersFrames(pcs)
+	for {
+		frame, more := frames.Next()
+		if strings.Contains(frame.File, gd2Repo) && !strings.Contains(frame.File, "vendor") {
+			entry.Data[SourceField] = fmt.Sprintf("[%s:%s:%d]", path.Base(frame.File), path.Base(frame.Function), frame.Line)
```
Comment (kshlm) [inline pkg/logging/hook.go]: Could you have the format be <file>:<line>:<function>. This would match glusterfs logs.

Comment (prashanthpai) [reply pkg/logging/hook.go]: Sure.

**Comment by prashanthpai** [comment]
glusterd1's log

```log
[2017-12-04 06:49:53.342553] W [MSGID: 103071] [rdma.c:4597:__gf_rdma_ctx_create] 0-rpc-transport/rdma: rdma_cm event channel creation failed [No such device]
```

gd2 log:
```log
[2017-12-04 12:34:04.020004] DEBUG [program.go:45:sunrpc.registerProgram]: registering sunrpc procedure procId={ProgramNumber:34123456 ProgramVersion:1 ProcedureNumber:1} procName=PortByBrick
[2017-12-04 12:31:22.894466]  INFO [server.go:143:sunrpc.(*SunRPC).Serve]: sunrpc client connected address=192.168.56.25:49149
```

**Pull request merged**